kernel/src/sched.rs

```diff
@@ -29,6 +29,10 @@ pub struct Kernel {
     work: Cell<usize>,
     /// This holds a pointer to the static array of Process pointers.
     processes: &'static [Option<&'static dyn process::ProcessType>],
+    /// Value to keep track of how many process identifiers have been used. Each
+    /// new process will get the next process identifier and this will be
+    /// incremented.
+    process_identifier_max: Cell<usize>,
```

Comment:
Maybe this should explicitly be a u64? `usize` is mostly going to be u32, and that's small enough to maybe overflow with enough process restarts.

Reply:
I can do that.